Installing pg_bm25 on M1 Macos

**Bug Description**
Trying to install pg_bm25 on macos M1 (Ventura) following the documentation for self-hosted postgres
Postgres15 installed with PostgresApp
paradedb = v0.2.18

**How To Reproduce**
Install postgres 15 using PostgresApp
Install pg_bm25

```
cargo pgrx init --pg15=`which pg_config`
cargo pgrx install
```

**Proposed Fix**
None

Trace:
```
RUST_BACKTRACE=full cargo pgrx install
       Using PgConfig("pg15") and `pg_config` from /Applications/Postgres.app/Contents/Versions/15/bin/pg_config
    Building extension with features pg15
     Running command "/Users/user/.rustup/toolchains/stable-aarch64-apple-darwin/bin/cargo" "build" "--features" "pg15" "--no-default-features" "--message-format=json-render-diagnostics"
    Finished dev [unoptimized + debuginfo] target(s) in 0.12s
  Installing extension
     Copying control file to /Applications/Postgres.app/Contents/Versions/15/share/postgresql/extension/pg_bm25.control
     Copying shared library to /Applications/Postgres.app/Contents/Versions/15/lib/postgresql/pg_bm25.so
 Discovering SQL entities
Error:
   0: couldn't parse postmaster
   1: Unsupported file format

Location:
   /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-pgrx-0.9.8/src/command/schema.rs:499

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ SPANTRACE ━━━━━━━━━━━━━━━━━━$$ ));━━━━━━━━━━━━━━━━━

   0: cargo_pgrx::command::schema::create_stub with postmaster_path=/Applications/Postgres.app/Contents/Versions/15/bin/postmaster postmaster_stub_dir=/Users/user/.pgrx/postmaster_stubs/Applications/Postgres.app/Contents/Versions/15/bin
      at /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-pgrx-0.9.8/src/command/schema.rs:460
       458 │ }
       459 │
       460 > #[tracing::instrument(level = "error", skip_all, fields(
       461 │     postmaster_path = %format_display_path(postmaster_path.as_ref())?,
       462 │     postmaster_stub_dir = %format_display_path(postmaster_stub_dir.as_ref())?,
   1: cargo_pgrx::command::schema::generate_schema with pg_version=15.4 profile=Dev test=false path=/Applications/Postgres.app/Contents/Versions/15/share/postgresql/extension/pg_bm25--0.0.0.sql features=["pg15"]
      at /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-pgrx-0.9.8/src/command/schema.rs:172
       170 │ }
       171 │
       172 > #[tracing::instrument(level = "error", skip_all, fields(
       173 │     pg_version = %pg_config.version()?,
       174 │     profile = ?profile,
   2: cargo_pgrx::command::install::install_extension with pg_version=15.4 profile=Dev test=false features=["pg15"]
      at /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-pgrx-0.9.8/src/command/install.rs:95
        93 │ }
        94 │
        95 > #[tracing::instrument(skip_all, fields(
        96 │     pg_version = %pg_config.version()?,
        97 │     profile = ?profile,
   3: cargo_pgrx::command::install::execute
      at /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-pgrx-0.9.8/src/command/install.rs:52
        50 │
        51 │ impl CommandExecute for Install {
        52 >     #[tracing::instrument(level = "error", skip(self))]
        53 │     fn execute(mut self) -> eyre::Result<()> {
        54 │         let metadata = crate::metadata::metadata(&self.features, self.manifest_path.as_ref())

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                                ⋮ 3 frames hidden ⋮
   4: cargo_pgrx::command::schema::create_stub::h6c2dba0a8608c580
      at <unknown source file>:<unknown line>
   5: cargo_pgrx::command::schema::generate_schema::h664e1aa730589c57
      at <unknown source file>:<unknown line>
   6: cargo_pgrx::command::install::copy_sql_files::h8dc5e319431573b5
      at <unknown source file>:<unknown line>
   7: cargo_pgrx::command::install::install_extension::haff2b95cbaeb6436
      at <unknown source file>:<unknown line>
   8: <cargo_pgrx::command::install::Install as cargo_pgrx::CommandExecute>::execute::h2d967cc161e6a4a8
      at <unknown source file>:<unknown line>
   9: <cargo_pgrx::command::pgrx::CargoPgrxSubCommands as cargo_pgrx::CommandExecute>::execute::hf205c93903f4883f
      at <unknown source file>:<unknown line>
  10: cargo_pgrx::main::h3192ef0d13e231cc
      at <unknown source file>:<unknown line>
  11: std::sys_common::backtrace::__rust_begin_short_backtrace::h53a0dfd737626df1
      at <unknown source file>:<unknown line>
  12: std::rt::lang_start::{{closure}}::h43eb9095ed9f97f8
      at <unknown source file>:<unknown line>
  13: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h7fff7c88c61df16f
      at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/ops/function.rs:284
  14: std::panicking::try::do_call::h5c3526811781b4da
      at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:502
  15: std::panicking::try::h64cb573047f67390
      at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:466
  16: std::panic::catch_unwind::h0ee316aa6f14634c
      at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panic.rs:142
  17: std::rt::lang_start_internal::{{closure}}::hbd896cc316ba7338
      at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/rt.rs:148
  18: std::panicking::try::do_call::hbae6886803fae47e
      at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:502
  19: std::panicking::try::hcc0fbb3995231d31
      at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:466
  20: std::panic::catch_unwind::hf36acbaf210582be
      at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panic.rs:142
  21: std::rt::lang_start_internal::hadaf077a6dd0140b
      at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/rt.rs:148
  22: _main<unknown>
      at <unknown source file>:<unknown line>

Run with COLORBT_SHOW_HIDDEN=1 environment variable to disable frame filtering.
```

Still have an issue but different

```
pg_bm25 % cargo pgrx install
       Using PgConfig("pg15") and `pg_config` from /opt/homebrew/opt/postgresql@15/bin/pg_config
    Building extension with features pg15
     Running command "/Users/user/.rustup/toolchains/stable-aarch64-apple-darwin/bin/cargo" "build" "--features" "pg15" "--no-default-features" "--message-format=json-render-diagnostics"
   Compiling pgrx-pg-sys v0.9.8
error: failed to run custom build command for `pgrx-pg-sys v0.9.8`

Caused by:
  process didn't exit successfully: `/Users/user/Documents/Sources/paradedb/target/debug/build/pgrx-pg-sys-312131b5860be2a6/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=PGRX_BUILD_VERBOSE
  cargo:rerun-if-env-changed=PGRX_PG_SYS_GENERATE_BINDINGS_FOR_RELEASE
  cargo:rerun-if-env-changed=PGRX_PG_CONFIG_PATH
  cargo:rerun-if-env-changed=PGRX_PG_CONFIG_AS_ENV
  cargo:rerun-if-env-changed=LLVM_CONFIG_PATH
  cargo:rerun-if-env-changed=LIBCLANG_PATH
  cargo:rerun-if-env-changed=LIBCLANG_STATIC_PATH
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64-apple-darwin
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64_apple_darwin
  cargo:rerun-if-env-changed=PGRX_PG_SYS_GENERATE_BINDINGS_FOR_RELEASE
  cargo:rerun-if-changed=include
  cargo:rerun-if-changed=cshim
  cargo:rerun-if-changed=/Users/user/.pgrx/config.toml
  cargo:rerun-if-env-changed=PGRX_PG_SYS_GENERATE_BINDINGS_FOR_RELEASE
  cargo:rerun-if-env-changed=PGRX_TARGET_INFO_PATH_PG15_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PGRX_TARGET_INFO_PATH_PG15
  cargo:rerun-if-env-changed=PGRX_INCLUDEDIR_SERVER_PG15_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PGRX_INCLUDEDIR_SERVER_PG15
  cargo:rerun-if-env-changed=PGRX_INCLUDEDIR_SERVER_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PGRX_INCLUDEDIR_SERVER
  cargo:rerun-if-env-changed=PGRX_BINDGEN_NO_DETECT_INCLUDES_aarch64-apple-darwin
  cargo:rerun-if-env-changed=PGRX_BINDGEN_NO_DETECT_INCLUDES

  --- stderr
  build_paths=BuildPaths { manifest_dir: "/Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pgrx-pg-sys-0.9.8", out_dir: "/Users/user/Documents/Sources/paradedb/target/debug/build/pgrx-pg-sys-5cb11818365f8f32/out", src_dir: "/Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pgrx-pg-sys-0.9.8/src", shim_src: "/Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pgrx-pg-sys-0.9.8/cshim", shim_dst: "/Users/user/Documents/Sources/paradedb/target/debug/build/pgrx-pg-sys-5cb11818365f8f32/out/cshim" }
  Generating bindings for pg15
  /opt/homebrew/opt/postgresql@15/include/postgresql/server/c.h:75:10: fatal error: 'libintl.h' file not found
  Error: bindgen failed for pg15

  Caused by:
     0: Unable to generate bindings for pg15
     1: clang diagnosed error: /opt/homebrew/opt/postgresql@15/include/postgresql/server/c.h:75:10: fatal error: 'libintl.h' file not found


  Location:
      /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pgrx-pg-sys-0.9.8/build.rs:753:10
 ```

--- a/.github/workflows/lint-rust.yml
+++ b/.github/workflows/lint-rust.yml
@@ -38,7 +38,7 @@ jobs:
           sudo apt-get update && sudo apt-get install -y postgresql-15 postgresql-server-dev-15
 
       - name: Install pgrx
-        run: cargo install cargo-pgrx --version 0.11.0
+        run: cargo install --locked cargo-pgrx --version 0.11.0
 
       - name: Initialize pgrx for Current PostgreSQL Version
         run: cargo pgrx init --pg15=/usr/lib/postgresql/15/bin/pg_config

--- a/.github/workflows/publish-pg_bm25.yml
+++ b/.github/workflows/publish-pg_bm25.yml
@@ -59,7 +59,7 @@ jobs:
           echo "/usr/lib/postgresql/${{ matrix.pg_version }}/bin" >> $GITHUB_PATH
 
       - name: Install pgrx
-        run: cargo install cargo-pgrx --version 0.11.0
+        run: cargo install --locked cargo-pgrx --version 0.11.0
 
       - name: Initialize pgrx for Current PostgreSQL Version
         working-directory: pg_bm25/

--- a/.github/workflows/publish-pg_search.yml
+++ b/.github/workflows/publish-pg_search.yml
@@ -59,7 +59,7 @@ jobs:
           echo "/usr/lib/postgresql/${{ matrix.pg_version }}/bin" >> $GITHUB_PATH
 
       - name: Install pgrx
-        run: cargo install cargo-pgrx --version 0.11.0
+        run: cargo install --locked cargo-pgrx --version 0.11.0
 
       - name: Initialize pgrx for Current PostgreSQL Version
         working-directory: pg_search/

--- a/.github/workflows/publish-third-party-pg_extensions.yml
+++ b/.github/workflows/publish-third-party-pg_extensions.yml
@@ -155,7 +155,7 @@ jobs:
           echo "/usr/lib/postgresql/${{ matrix.pg_version }}/bin" >> $GITHUB_PATH
 
       - name: Install pgrx
-        run: cargo install cargo-pgrx --version 0.11.0
+        run: cargo install --locked cargo-pgrx --version 0.11.0
 
       - name: Initialize pgrx for Current PostgreSQL Version
         working-directory: pgml-extension/

--- a/.github/workflows/test-pg_bm25.yml
+++ b/.github/workflows/test-pg_bm25.yml
@@ -49,7 +49,7 @@ jobs:
 
       - name: Install pgrx, grcov & llvm-tools-preview
         run: |
-          cargo install cargo-pgrx --version 0.11.0 && cargo install grcov
+          cargo install --locked cargo-pgrx --version 0.11.0 && cargo install --locked grcov
           rustup component add llvm-tools-preview
 
       - name: Initialize pgrx for Current PostgreSQL Version

--- a/.github/workflows/test-pg_search.yml
+++ b/.github/workflows/test-pg_search.yml
@@ -49,7 +49,7 @@ jobs:
 
       - name: Install pgrx, grcov & llvm-tools-preview
         run: |
-          cargo install cargo-pgrx --version 0.11.0 && cargo install grcov
+          cargo install --locked cargo-pgrx --version 0.11.0 && cargo install --locked grcov
           rustup component add llvm-tools-preview
 
       - name: Initialize pgrx for Current PostgreSQL Version

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -121,7 +121,7 @@ RUN apt-fast update && apt-fast install -y --no-install-recommends \
 ENV PATH="/root/.cargo/bin:$PATH" \
     PGX_HOME=/usr/lib/postgresql/${PG_VERSION_MAJOR}
 
-RUN cargo install cargo-pgrx --version 0.11.0 && \
+RUN cargo install --locked cargo-pgrx --version 0.11.0 && \
     cargo pgrx init "--pg${PG_VERSION_MAJOR}=/usr/lib/postgresql/${PG_VERSION_MAJOR}/bin/pg_config"
 
 ######################

--- a/pg_bm25/README.md
+++ b/pg_bm25/README.md
@@ -182,11 +182,18 @@ Please refer to the [documentation](https://docs.paradedb.com/search/bm25) for a
 Before developing the extension, ensure that you have Rust installed
 (version >1.70), ideally via `rustup` (we've observed issues with installing Rust via Homebrew on macOS).
 
+If you are on macOS and using Postgres.app, you'll first need to add the `pg_config` binary to your path:
+
+```bash
+export PATH="$PATH:/Applications/Postgres.app/Contents/Versions/latest/bin"
+```
+
 Then, install and initialize pgrx:
 
 ```bash
-cargo install cargo-pgrx --version 0.11.0
-cargo pgrx init
+# Note: Replace --pg15 with your version of Postgres, if different (i.e. --pg16, --pg14, etc.)
+cargo install --locked cargo-pgrx --version 0.11.0
+cargo pgrx init --pg15=`which pg_config`
 ```
 
 ### Running the Extension

--- a/pg_search/README.md
+++ b/pg_search/README.md
@@ -116,11 +116,18 @@ Please refer to the [documentation](https://docs.paradedb.com/search/hybrid) for
 Before developing the extension, ensure that you have Rust installed
 (version >1.70), ideally via `rustup` (we've observed issues with installing Rust via Homebrew on macOS).
 
+If you are on macOS and using Postgres.app, you'll first need to add the `pg_config` binary to your path:
+
+```bash
+export PATH="$PATH:/Applications/Postgres.app/Contents/Versions/latest/bin"
+```
+
 Then, install and initialize pgrx:
 
 ```bash
-cargo install cargo-pgrx --version 0.11.0
-cargo pgrx init
+# Note: Replace --pg15 with your version of Postgres, if different (i.e. --pg16, --pg14, etc.)
+cargo install --locked cargo-pgrx --version 0.11.0
+cargo pgrx init --pg15=`which pg_config`
 ```
 
 ### Running the Extension

